drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -1268,6 +1268,10 @@ static int mmcsd_transferready(FAR struct mmcsd_state_s *priv)
           goto errorout;
         }
 
+      /* Do not hog the CPU */
+
+      nxsig_usleep(1000);
```

Comment:
This code is executed on the next operation after a write that has ben committed but not completed by the card yet. So there is some delay consumed by the cpu running the time to the next cycle.  
Write--------------------------------V Completion
Next(a)--------------------------V short wait
Next(b)-------------------------------------V wait no wait

> In other words: what is the usual time until transmission is complete?

This is usually short 1 ms to 250 MS or longer on some cards during wear-leveling

If thruput is an issue and system performance matters in the sense you do not want to busy wait.  Then  CONFIG_MMCSD_SDIOWAIT_WRCOMPLETE  should be used. The CPU will wait on a semaphore for the completion interrupt from D0 and not busy wait. On the D0 completion,  the loop will read the status and exit. With this setting there is 0 impact on thruput this has been tested with sd_bench.  In the case of an SD failure it may wait for 1 Second. But now it will not a busy wait.

If CONFIG_MMCSD_SDIOWAIT_WRCOMPLETE is not set.  Then nxsig_usleep time will be at the mercy of the tick and rounding. If we just yield we are also at the the mercy of the tick and task load. 

Do you feel the 100 is really an advantage over 1000 on a 10 ms system, will it be on a 1 ms with quantization?

Reply:
can we change CONFIG_MMCSD_SDIOWAIT_WRCOMPLETE from bool to int? so the user could select the different sleep time.

Reply:
I'm fine with 1ms, just felt I need to ask it since you are more familiar with that part. We can go with Kconfig value as well, but I'm not sure if that will give us significant benefit.

Reply:
> can we change CONFIG_MMCSD_SDIOWAIT_WRCOMPLETE from bool to int? so the user could select the different sleep time.

No - it is a separate setting.

Reply:
> I'm fine with 1ms, just felt I need to ask it since you are more familiar with that part. We can go with Kconfig value as well, but I'm not sure if that will give us significant benefit.

Let's Take in as is. We can tune it if need be later.

Reply:
@pkarashchenko @xiaoxiang781216 can we bring this in or do you want more changes?

Reply:
I am fine with the current state.